src/SourceBuild/content/prep.sh

```diff
@@ -107,22 +111,32 @@ fi
 function DownloadArchive {
   archiveType="$1"
   isRequired="$2"
+  artifacts="$3"
 
   packageVersionsPath="$SCRIPT_ROOT/eng/Versions.props"
   notFoundMessage="No source-built $archiveType found to download..."
 
   echo "  Looking for source-built $archiveType to download..."
-  archiveVersionLine=$(grep -m 1 "<PrivateSourceBuilt${archiveType}Url>" "$packageVersionsPath" || :)
-  versionPattern="<PrivateSourceBuilt${archiveType}Url>(.*)</PrivateSourceBuilt${archiveType}Url>"
+  archiveVersionLine=$(grep -m 1 "<SourceBuiltVersionNumber>" "$packageVersionsPath" || :)
+  versionPattern="<SourceBuiltVersionNumber>(.*)</SourceBuiltVersionNumber>"
   if [[ $archiveVersionLine =~ $versionPattern ]]; then
-      archiveUrl="${BASH_REMATCH[1]}"
+    archiveVersion="${BASH_REMATCH[1]}"
+    archiveUrl="https://dotnetcli.azureedge.net/source-built-artifacts/assets/Private.SourceBuilt.$archiveType.$archiveVersion.$artifacts.tar.gz"
```

Comment:
This is not going to work for the prebuilts because there is typically one version for all rids.

Reply:
You're saying that for alpine, we would still pull the centos8 prebuilts?

Reply:
[Modified to check for the archiveType](https://github.com/dotnet/installer/pull/17370/files#diff-a1d21cbd7f180cea469d4fb634d555679ca532f21d903b2b5df311f6a051aa6aR127)